external_libraries/mpi_python/mpi_python.h

```diff
@@ -284,12 +285,98 @@ class PythonMPI
         return result_val;
     }
 
+	/// Perform a MPI_Scatter operation.
+	/**
+	 * Scattering a vector of values from the RankToScatterFrom process.
+	 * @param rComm A communicator object.
+	 * @param rLocalValues The local values to be scattered.
+	 * @param RankToScatterFrom The MPI rank of the process where the values will be scattered from.
+	 * @return The local value that was scattered
+	 */
+    template<class TValueType>
+	TValueType scatter(PythonMPIComm& rComm,
+                       const std::vector<TValueType>& rLocalValues,
+                       const int RankToScatterFrom)
+    {
+        // Determime data type
+        const MPI_Datatype DataType = this->GetMPIDatatype(TValueType());
+        int rank, size;
+        MPI_Comm_rank(rComm.GetMPIComm(), &rank);
+        MPI_Comm_size(rComm.GetMPIComm(), &size);
+        if (rank == RankToScatterFrom && rLocalValues.size() != size)
+            throw std::runtime_error("Wrong number of values to Scatter!");
+        TValueType receive_val;
+        // Communicate
+        MPI_Scatter(rLocalValues.data(), 1, DataType, &receive_val,
```

Comment:
You might want to check the return value of MPI_Scatter for success or failure.

Reply:
good idea!

Reply:
I checked how to do this but I could not find a clear explanation
Scatter does not accept Status and Request as argument

Also I read that it is not necessary to check for the return value, since errors will by default lead to a termination of the execution

Reply:
Might be, I am not sure

However if we start introducing this then I think we should do it consistently across Kratos